### ISSUE TITLE
Why do I still need @ThreadLeakFilters for threads named "elasticsearch"?

### ISSUE THREAD

**Issue by mikerott** [elastic/elasticsearch#12251]
I recently updated from 1.1.1 (yeah, I know) to 1.5.2.  The easy parts were changing some of the values in `@ClusterScope` annotation and a few method signature changes here and there.  But then I hit a blocker.  I'm using a class that `extends ElasticsearchIntegrationTest`.  Against 1.1.1, I had no need for `@ThreadLeakFilters` annotation, but now against 1.5.2, I find that I *do* need it so I can explicitly ignore threads named `elasticsearch`.

Without the annotation, I see the following failures during test execution:

```
com.carrotsearch.randomizedtesting.ThreadLeakError: 65 threads leaked from SUITE scope at com.mike.IndexManagerTest: 
   1) Thread[id=124, name=elasticsearch[Cypher][http_server_worker][T#14]{New I/O worker #74}, state=RUNNABLE, group=TGRP-IndexManagerTest]
        at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
        at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:159)
        at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:103)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
        at org.elasticsearch.common.netty.channel.socket.nio.SelectorUtil.select(SelectorUtil.java:68)
        at org.elasticsearch.common.netty.channel.socket.nio.AbstractNioSelector.select(AbstractNioSelector.java:434)
        at org.elasticsearch.common.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:212)
        at org.elasticsearch.common.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89)
... [63 more]
```

So I added the annotation to my class that `extends ElasticsearchIntegrationTest`:

```
@ThreadLeakFilters(defaultFilters = true, filters = {TestRunnerThreadsFilter.class})
```

And the TestRunnerThreadsFilter class is:

```
import com.carrotsearch.randomizedtesting.ThreadFilter;

public class TestRunnerThreadsFilter implements ThreadFilter {

	@Override
	public boolean reject(Thread thread) {
		// randomizedtesting shouldn't bother checking for zombie threads for these:
		String threadName = thread.getName();

		if (threadName.startsWith("elasticsearch")) {
			return true;
		}
		
		return false;
	}

}
```

Why do I need to do this?  Do I have a more fundamental problem I need to investigate?

**Comment by bleskes** [comment]
It seems to imply cluster / nodes are not properly shut down after the suite ends.   Any idea why that would happen? do you override any of the setup / teardown methods?

**Comment by s1monw** [comment]
in `1.1.x` we did filter out all those threads ourself. Now in 1.5 we shutdown stuff properly. Can you provide one test that leaks threads so we can take a look?

**Comment by mikerott** [comment]
I pushed a recreate to https://github.com/mikerott/es_issue_12251

**Comment by mikerott** [comment]
Adding super() (or equivalent) to all `@Before`, `@BeforeClass`, `@After`, `@AfterClass` does not resolve the need for `@ThreadLeakFilters`.

**Comment by clintongormley** [comment]
Sorry it has been a while @mikerott - is this still an issue?

**Comment by clintongormley** [comment]
Closed by #17921
